File: NodeJS (Front)/src/services/guiaRemisionService.js
import api from './api'

const guiaRemisionService = {
  buscar: (idOrdenCompra) =>
    api.get('/guia-remision', { params: { idOrdenCompra } }),

  listarDetalle: (idGuiaRemision) =>
    api.get(`/guia-remision/${idGuiaRemision}/detalle`),

  crear: (idOrdenCompra, idUsuario, numeroGuia) =>
    api.post('/guia-remision', { idOrdenCompra, idUsuario, numeroGuia }),

  agregarDetalle: (idGuiaRemision, detalle) =>
    api.post(`/guia-remision/${idGuiaRemision}/detalle`, detalle),

  anular: (idGuiaRemision) =>
    api.patch(`/guia-remision/${idGuiaRemision}/anular`),
}

export default guiaRemisionService

File: SistemaBodegaJaime/BodegaJaimeBackend/src/main/java/com/proyectosdejava/service/GuiaRemisionService.java
package com.proyectosdejava.service;

import com.proyectosdejava.model.DetalleGuiaRemisionDTO;
import com.proyectosdejava.model.GuiaRemisionDTO;
import com.proyectosdejava.repository.GuiaRemisionRepository;
import jakarta.transaction.Transactional;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Service
public class GuiaRemisionService {

    @Autowired
    private GuiaRemisionRepository guiaRemisionRepository;

    public List<GuiaRemisionDTO> buscar(Integer idOrdenCompra) {
        return guiaRemisionRepository.buscar(idOrdenCompra);
    }

    public List<DetalleGuiaRemisionDTO> listarDetalle(Integer idGuiaRemision) {
        return guiaRemisionRepository.listarDetalle(idGuiaRemision);
    }

    @Transactional
    public Map<String, Object> crear(Integer idOrdenCompra, Integer idUsuario, String numeroGuia) {
        Object[] resultado = guiaRemisionRepository.crear(idOrdenCompra, idUsuario, numeroGuia);
        Map<String, Object> respuesta = new HashMap<>();
        respuesta.put("idGenerado", resultado[0]);
        respuesta.put("mensaje", resultado[1]);
        return respuesta;
    }

    @Transactional
    public Map<String, String> agregarDetalle(Integer idGuiaRemision, Integer idProducto, Integer cantidad) {
        return buildRespuesta(guiaRemisionRepository.agregarDetalle(idGuiaRemision, idProducto, cantidad));
    }

    @Transactional
    public Map<String, String> anular(Integer idGuiaRemision) {
        return buildRespuesta(guiaRemisionRepository.anular(idGuiaRemision));
    }

    private Map<String, String> buildRespuesta(String resultado) {
        Map<String, String> respuesta = new HashMap<>();
        respuesta.put("mensaje", resultado);
        return respuesta;
    }
}